lib/stdlib/doc/src/peer.xml

```diff
@@ -526,11 +526,81 @@
-        stopping may be graceful (via <c>init:stop()</c>) or less so,
-        by just closing the control connection, which triggers <c>erlang:halt()</c>
-        executed by the peer.</p>
-        <p>A graceful shutdown can be slower, or even block indefinitely.</p>
+        <p>
+          Stops a peer node. How the node is stopped depends on the
+          <seetype marker="#start_options"><c>shutdown</c></seetype>
+          option passed when starting the peer node. Currently the
+          following <c>shutdown</c> options are supported:
+        </p>
+        <taglist>
+          <tag><c>halt</c></tag>
+          <item><p>
+            This is the default shutdown behaviour. It behaves as <c>shutdown</c>
+            option <c>{halt, DefaultTimeout}</c> where <c>DefaultTimeout</c>
+            currently equals <c>5000</c>.
+          </p></item>
+          <tag><c>{halt, Timeout}</c></tag>
+          <item><p>
+            Triggers a call to
+            <seemfa marker="erts:erlang#halt/0"><c>erlang:halt()</c></seemfa>
+            on the peer node and then waits for the Erlang distribution
+            connection to the peer node to be taken down. If this connection
+            has not been taken down after <c>Timeout</c> milliseconds, it will
+            forcefully be taken down by <c>peer:stop/1</c>. See the
+            <seeerl marker="#dist_connection_close">warning</seeerl> below for
+            more info about this.
+          </p></item>
+          <tag><c>Timeout</c></tag>
+          <item><p>
+            Triggers a call to
+            <seemfa marker="erts:init#stop/0"><c>init:stop()</c></seemfa>
+            on the peer node and then waits for the Erlang distribution
+            connection to the peer node to be taken down. If this connection
+            has not been taken down after <c>Timeout</c> milliseconds, it will
+            forcefully be taken down by <c>peer:stop/1</c>. See the
+            <seeerl marker="#dist_connection_close">warning</seeerl> below for
+            more info about this.
+          </p></item>
+          <tag><c>close</c></tag>
+          <item><p>
+            Close the <i>control connection</i> to the peer node and
+            return. This is the fastest way for the caller of
+            <c>peer:stop/1</c> to stop a peer node.
+          </p>
+          <p>
+            Note that if the Erlang distribution connection is not used as
+            control connection it might not have been taken down when
+            <c>peer:stop/1</c> returnes. Also note that the
+            <seeerl marker="#dist_connection_close">warning</seeerl> below
+            applies when the Erlang distribution connection is used as control
+            connection.
+          </p>
+          </item>
+        </taglist>
+
+        <p>
+          Currently integer <c>Timeout</c> values less than <c>1000</c> will
+          silently be increased to <c>1000</c>, and integer <c>Timeout</c>
```

Comment:
To be honest, I'd rather have `verify_args` to throw when an invalid (<1000 or > max 32 bit int) is passed. Silently correcting invalid argument tends to swallow errors instead of surfacing them early (in `peer:start_link`).

Leaving the decision to you, but I introduced `verify_args` specifically to address that (since it was too easy to pass some invalid node names and then waste a lot of time trying to figure out what's wrong with that peer).

Reply:
I've changed it to fail in `verify_args`